tests/run-pass/issue-58645.rs

```diff
@@ -0,0 +1,5 @@
+fn main() {
+    unsafe {
+        std::ptr::write_volatile(&mut (0, 0), (0, 0));
```

Comment:
Please expand the test to write to some place that we can read back from later, and compare the result using `assert_eq!`.